src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -9655,6 +9650,12 @@ int RGWRados::cls_bucket_list_ordered(const DoutPrefixProvider *dpp,
     num_entries << " total entries" << dendl;
 
   auto& ioctx = index_pool;
+
+  // XXX: check_disk_state() relies on ioctx.get_last_version() but that
+  // returns 0 because CLSRGWIssueBucketList doesn't make any synchonous calls
```

Comment:
this has been broken for ordered listings ever since CLSRGWIssueBucketList was added (10 years ago with https://github.com/ceph/ceph/commit/6f44f7a0a9847e419ce2783164633efa71218380)

to make this work with sharded buckets, i think we'd need to track the versions for each shard object, and pass in the correct shard's version when sending dir suggestions